### DISCUSSION TITLE
How to obtain the file name of the current activity file, why do I always obtain the value of untitled

### DISCUSSION THREAD

**Discussion by likemuuxi** [SilentVoid13/Templater#1351] [Help]
The function I want to achieve is:
Obtain the path and file name of the current activity file, and create a new path consisting of "current file path"+"/"+"current activity file name"+"/"
Then pop up the prompt box, enter a new file name and store it in a variable, then rename the file and move it to the new file path

However, the problem is that I always get the name of the current active file as untitled
```
<%*
const pathVar = tp.file.folder(true) + "/" + tp.file.title + "/"
let newTitle = await tp.system.prompt(pathVar);
if (newTitle == null || newTitle == "未命名") { // 用户取消输入或者输入为空
    newTitle = tp.file.title; // 保持原来的文件名
}
await tp.file.create_new("MyFileContent", newTitle, false, app.vault.getAbstractFileByPath(pathVar))
-%>
```

**Comment by Zachatoo** [comment]
Which Templater command are you using to trigger the template? If you're using the "Templater: Create new note from template" command, then Templater will create an "Untitled" note for you, then insert and execute the template into that note. In that case, `tp.file.title` is correctly "Untitled".

For this template, since you are explicitly creating a file using `tp.file.create_new()`, you probably want to use one of the other Templater commands to insert a template, then `tp.file.title` will be the active file name.

**Reply by likemuuxi** [reply]
I have tried but failed, and I have chosen to give up this method. Thank you for your help

**Reply by likemuuxi** [reply]
I have studied the following topic and I am unable to reproduce it. Is it because something has changed？https://github.com/SilentVoid13/Templater/discussions/1000
I will obtain the following two files "Untitled", "Untitled-address"
